model_eval.py

```diff
@@ -71,9 +105,44 @@ def train_model(self):
 
         trainer.save_model("results/models")
 
+        if self.task == "dep":
+            return model_dep, model_heads, dataset_dict, labels_mapper_dep, labels_mapper_head
+    
         return model, dataset_dict, labels_mapper
 
     def evaluate_model(self):
+        if self.task == "dep":
+            trained_model_dep, trained_model_heads, dataset_dict, labels_mapper_dep, labels_mapper_head = self.train_model()
+            trained_model_dep.eval()
+
+            reverse_label_map_dep = {v: k for k, v in labels_mapper_dep.items()}
+            reverse_label_map_head = {v: k for k, v in labels_mapper_head.items()}
+
+            pred_dep_labels = collect_predictions_dep(dataset_dict["dep"]["test"], 
+                                                      trained_model_dep, 
+                                                      reverse_label_map_dep,
+                                                      self.device)
+
+            gold_dep_labels = collect_gold_labels_dep(dataset_dict["dep"]["test"]["labels"],
+                                                      reverse_label_map_dep)
+
+            pred_head_labels = collect_predictions_dep(dataset_dict["dep"]["test"], 
+                                                       trained_model_heads, 
+                                                       reverse_label_map_head ,
+                                                       self.device)
+            
+            gold_head_labels =  collect_gold_labels_dep(dataset_dict["dep"]["test"]["labels"],
+                                                        reverse_label_map_head)
+            
+            words_test_set = extract_dep_data("data/genia/test.txt") # need to define the correct path
+            words_test_set = [item[0] for sample in words_test_set for item in sample]
+
+            uas, las = calculate_uas_las(words_test_set, 
+                                         gold_head_labels, 
+                                         gold_dep_labels, 
+                                         pred_head_labels, 
+                                         pred_dep_labels)
+
         trained_model, dataset_dict, labels_mapper = self.train_model()

```

Comment:
Everything below here should be in an else statement unless a return statement is located at the end of the previous if statement in line 114.